File: mobile-client/android/UI/Message/Thread/MessageThreadScreenContract.kt
package app.mobiling.client.ui.message.thread

import app.mobiling.client.contract.message.thread.MessageItemPayload
import app.mobiling.client.contract.message.thread.MessageThreadSummary

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
data class MessageThreadIndexState(
    val threads: List<MessageThreadSummary> = emptyList(),
)

data class MessageThreadDetailState(
    val threadId: String,
    val items: List<MessageItemPayload> = emptyList(),
)

File: mobile-client/ios/UI/Message/Thread/MessageThreadScreenContract.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public struct MessageThreadIndexState: Sendable {
    public let threads: [MessageThreadSummary]

    public init(threads: [MessageThreadSummary] = []) {
        self.threads = threads
    }
}

public struct MessageThreadDetailState: Sendable {
    public let threadId: String
    public let items: [MessageItemPayload]

    public init(threadId: String, items: [MessageItemPayload] = []) {
        self.threadId = threadId
        self.items = items
    }
}
